lib/SelectionBar/index.js

```diff
@@ -1,80 +1,47 @@
-import React, { Fragment } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
 import Row from 'react-bootstrap/lib/Row';
-import Col from 'react-bootstrap/lib/Col';
-import SelectionBarInformation from './SelectionBarInformation';
-import SelectionBarCancel from './SelectionBarCancel';
 import SelectionBarAction from './SelectionBarAction';
 import SelectionBarDivider from './SelectionBarDivider';
+import SelectionBarCancel from './SelectionBarCancel';
+import SelectionBarInformation from './SelectionBarInformation';
+import SelectionBarActions from './SelectionBarActions';
+import SelectionBarCounter from './SelectionBarCounter';
 
 const SelectionBar = props => {
-  const { length } = props.selectedIds;
-  const hasSelected = length !== 0;
   const classNames = cx('selection-bar', {
     'selection-bar--fixed': props.fixed,
     'selection-bar--auto-hide': props.autoHide,
-    'has-selected': hasSelected
+    'has-selected': props.hasSelected
   });
+
   return (
     <Row className={classNames}>
-      <Col
-        xs={12}
-        sm={4}
-        md={3}
-        className="selection-bar__section selection-bar__information"
-      >
-        <SelectionBarInformation
-          length={length}
-          type={props.type}
-          selectAll={props.selectAll}
-        />
-      </Col>
-      {hasSelected && (
-        <Fragment>
-          <Col
-            xs={12}
-            sm={5}
-            md={6}
-            className="selection-bar__section selection-bar__actions"
-          >
-            {props.children}
-          </Col>
-          <Col
-            xsHidden
-            sm={3}
-            md={3}
-            className="selection-bar__section selection-bar__cancel"
-          >
-            <SelectionBarCancel
-              hasSelected={hasSelected}
-              clearSelection={props.clearSelection}
-            />
-          </Col>
-        </Fragment>
-      )}
+      {props.children}
+      {props.onCancel && <SelectionBarCancel onCancel={props.onCancel} />}
     </Row>
   );
 };
 
+SelectionBar.Information = SelectionBarInformation;
 SelectionBar.Action = SelectionBarAction;
+SelectionBar.Actions = SelectionBarActions;
 SelectionBar.Divider = SelectionBarDivider;
+SelectionBar.Counter = SelectionBarCounter;
 
 SelectionBar.propTypes = {
-  type: PropTypes.string.isRequired,
-  selectedIds: PropTypes.arrayOf(
-    PropTypes.oneOfType([PropTypes.number, PropTypes.string])
-  ).isRequired,
+  hasSelected: PropTypes.bool.isRequired,
   children: PropTypes.node.isRequired,
-  clearSelection: PropTypes.func.isRequired,
-  selectAll: PropTypes.func.isRequired,
   fixed: PropTypes.bool,
-  autoHide: PropTypes.bool
+  autoHide: PropTypes.bool,
+  onCancel: PropTypes.func
 };
 
 SelectionBar.defaultProps = {
   fixed: false,
-  autoHide: false
+  autoHide: false,
+  onCancel: undefined
```

Comment:
💚What do you think to using null here instead of undefined? I think it makes more sense as null is an assigned value, whereas undefined suggests that the variable hasn't been defined, which it actually has.